Tests/MongoSwiftTests/AsyncTestUtils.swift

```diff
@@ -27,6 +28,16 @@ extension MongoClient {
                 opts.serverAPI!.version = apiVersion
             }
         }
+
```

Comment:
all the logic we're starting to repeat across the sync/async `makeTestClient` methods makes me think we should have some kind of common method to resolve client options and environment variables in `TestsCommon`? seems out of scope for this PR though

Reply:
Yeah that's definitely a good idea. I ran into a bunch of bugs over the course of implementing this due to differences between the two implementations. Filed SWIFT-1241 to cover the work for doing this.